Feature/user managed identity

logachev opened the pull request:
Internal PR to gather feedback

Commit:
Add userAssigned identity support for VM and VMSS

Commit:
Update VM and VMSS Identity tests

Commit:
Add msi user assigned identity package

Commit:
Add User Assigned Identity resource

Commit:
Add User Assigned Identity tests

Commit:
User assigned identity test for VMSS

Commit:
Add principal_id for user assigned identity

Commit:
Update docs

Commit:
Fix docs typos & move docs to the right place

wesyao commented on website/docs/r/user_assigned_identity.markdown:
```diff
@@ -0,0 +1,64 @@
+* `tags` - (Optional) A mapping of tags to assign to the resource.
+
+## Attributes Reference
+
```
possible example?

logachev replied:
Example is in a first section

wesyao commented on website/docs/r/user_assigned_identity.markdown:
```diff
@@ -0,0 +1,64 @@
+* `location` - (Required) The location/region where the user assigned identity is
+    created.
+
+* `tags` - (Optional) A mapping of tags to assign to the resource.
```
include example adding tags

logachev replied:
Most of components don't have it. Since Tags is a common thing, not smth unique for this resource I don't think we need it.

wesyao commented on website/docs/r/user_assigned_identity.markdown:
````diff
@@ -0,0 +1,64 @@
+User Assigned Identitites can be imported using the `resource id`, e.g.
+
+```shell
+terraform import azurerm_user_assigned_identity.testIdentity /subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/acceptanceTestResourceGroup1/providers/Microsoft.ManagedIdentity/userAssignedIdentities/testIdentity
````
i think it needs to be more clear what i need to fill in here

wesyao replied:
eg: /subscriptions/<subscription_id>/resourcegroups/<resource_group_name>

logachev replied:
What I used is common across Terraform docs, don't think we should change their format.

wesyao commented on website/docs/r/virtual_machine.html.markdown:
```diff
@@ -441,7 +441,9 @@ resource "azurerm_virtual_machine" "test" {
 `identity` supports the following:
 
-* `type` - (Required) Specifies the identity type of the virtual machine. The only allowable value is `SystemAssigned`. To enable Managed Service Identity the virtual machine extension "ManagedIdentityExtensionForWindows" or "ManagedIdentityExtensionForLinux" must also be added to the virtual machine. The Principal ID can be retrieved after the virtual machine has been created, e.g.
+* `type` - (Required) Specifies the identity type of the virtual machine. Allowable values are `SystemAssigned` and `UserAssigned`. To enable Managed Service Identity the virtual machine extension "ManagedIdentityExtensionForWindows" or "ManagedIdentityExtensionForLinux" must also be added to the virtual machine. For the `SystemAssigned` identity the Principal ID can be retrieved after the virtua ...
```
what's the difference between`SystemAssigned` and `UserAssigned`. it would be helpful to know what the difference is

logachev replied:
Added a link to MS docs.

wesyao commented on website/docs/r/virtual_machine_scale_set.html.markdown:
```diff
@@ -274,7 +274,9 @@ The following arguments are supported:
 `identity` supports the following:
 
-* `type` - (Required) Specifies the identity type to be assigned to the scale set. The only allowable value is `SystemAssigned`. To enable Managed Service Identity (MSI) on all machines in the scale set, an extension with the type "ManagedIdentityExtensionForWindows" or "ManagedIdentityExtensionForLinux" must also be added. The scale set's Service Principal ID (SPN) can be retrieved after the sca ...
+* `type` - (Required) Specifies the identity type to be assigned to the scale set. Allowable values are `SystemAssigned` and `UserAssigned`. To enable Managed Service Identity (MSI) on all machines in the scale set, an extension with the type "ManagedIdentityExtensionForWindows" or "ManagedIdentityExtensionForLinux" must also be added. For the `SystemAssigned` identity the scale set's Service Pri ...
```
same as above

logachev replied:
Added a link to MS docs.

Commit:
Add link to ms docs

Commit:
Merge latest from master branch

Commit:
Address review feedback

Commit:
Merge with latest master

Commit:
Update tests to limit UAI name to 24 symbols

Commit:
More review feedback

tombuildsstuff committed:
User Assigned Identities: refactoring mentioned in PR

```
acctests azurerm TestAccAzureRMUserAssignedIdentity_
=== RUN   TestAccAzureRMUserAssignedIdentity_importBasic
--- PASS: TestAccAzureRMUserAssignedIdentity_importBasic (80.67s)
=== RUN   TestAccAzureRMUserAssignedIdentity_basic
--- PASS: TestAccAzureRMUserAssignedIdentity_basic (78.45s)
PASS
ok  	github.com/terraform-providers/terraform-provider-azurerm/azurerm	159.156s
```
